src/arduino.cc/builder/phases/libraries_builder.go

```diff
@@ -93,7 +93,7 @@ func fixLDFLAGforPrecompiledLibraries(ctx *types.Context, libraries []*types.Lib
 				name = strings.Replace(name, "lib", "", 1)
 				libs_cmd += "-l" + name + " "
 			}
-			ctx.BuildProperties[constants.BUILD_PROPERTIES_COMPILER_C_ELF_EXTRAFLAGS] += "\"-L" + path + "\" " + libs_cmd
+			ctx.BuildProperties[constants.BUILD_PROPERTIES_COMPILER_LDFLAGS] += "\"-L" + path + "\" " + libs_cmd
```

Comment:
Hi @facchinm,

I am attempting to use the 1.9.x beta of the Arduino IDE to try and use precompiled libs. I ran into an issue,
I faced 2 problems here
- No whitespace between the last existing value of compiler.ldflags and the appended string. (-L ..., -lmylib). Example: using the samd core 1.6.17, the concatenation looks like ```'--warn-section-align"-LC:\Users\```. Adding spaces after ```--warn-section-align``` simply gets removed out.
- These flags should be at the very end, typically after -lm etc. otherwise they won't be utilized by the linker appropriately. Basically the order matters. More info on [stackoverflow](https://stackoverflow.com/questions/5005363/undefined-reference-to-sin).

Reply:
Hi @kgoveas ,
which core are you targeting? The flag is added only if {compiler.ldflags} is specified in `recipe.c.combine.pattern` ; both the "no space" issue and position should be targeted at core level on `platform.txt`.
If you could link the one you are using I can give you some advice :wink:

Reply:
Hi @facchinm,

I am using the Arduino samd core v1.6.17. Link to the line of [platform.txt](https://github.com/arduino/ArduinoCore-samd/blob/d58b840f63e78ce0b890891d5cfae92ba8090d26/platform.txt#L50)

Reply:
Ah right, we already had some reports about it https://github.com/arduino/Arduino/issues/7005#issuecomment-350547038 . 
The main problem is, in fact, the "abuse" of the existing `ldflags` variable; probably adopting another name would solve the issue for every core at the cost of modifying `recipe.c.combine.pattern` (which should be modified anyway due to the link order issue).
So probably using a brand new entry would allow a much cleaner implementation.
I'll prepare a PR to address it early next week.

Reply:
@facchinm, Unfortunate indeed. Also the comment a few lines above is false. Your PR changed it from extra_flags to ldflags. 

Also, would recommend to include a fix for https://github.com/arduino/arduino-builder/issues/256 by adding both build.arch and build.mcu as -L paths. That way specific MCU architectures like the cortex-m4 can be provided based on the build.arch parameter while the rest are based on build.mcu.

Thanks.